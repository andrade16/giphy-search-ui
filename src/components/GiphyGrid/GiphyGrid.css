.giphy-grid-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  overflow: hidden;
  background-color: #fff;
  height: 70vh;
  margin: 25px;
}

.grid-list {
  width: 600px;
  height: 450px;
}

.image-container {
  width: 100%;
  height: 100%;
}

.image-container > img {
  width: 100%;
  height: 100%;
}

img {
  width: 100%;
  height: 100%;
}
