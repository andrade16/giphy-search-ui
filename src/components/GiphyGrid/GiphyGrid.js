import React, { useRef } from "react";
import GridList from "@material-ui/core/GridList";
import GridListTile from "@material-ui/core/GridListTile";
import "./GiphyGrid.css";

const GiphyGrid = ({ gifs, isLoading }) => {
  const imgCounter = useRef(0);

  // keep track of when individual gifs are done loading.
  // until all gis per page are done loading, display a progress bar
  const imageLoaded = () => {
    imgCounter.current += 1;
    if (imgCounter.current >= gifs.length) {
      isLoading(false);
      imgCounter.current = 0;
    }
  };

  return (
    <div className="giphy-grid-container">
      <GridList className="grid-list" cellHeight={160} cols={3}>
        {gifs.map((gif) => (
          <GridListTile key={gif.id} cols={1}>
            <img
              alt={gif.title}
              src={gif.images.fixed_width.url}
              onLoad={imageLoaded}
            />
          </GridListTile>
        ))}
      </GridList>
    </div>
  );
};

export default GiphyGrid;
