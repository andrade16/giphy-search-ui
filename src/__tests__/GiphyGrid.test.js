import React from "react";
import renderer from "react-test-renderer";
import GiphyGrid from "../components/GiphyGrid/GiphyGrid";

const mockGifs = [
  {
    analytics: {},
    analytics_response_payload: "",
    bitly_gif_url: "https://gph.is/2pLKLNW",
    bitly_url: "https://gph.is/2pLKLNW",
    content_url: "",
    embed_url: "https://giphy.com/embed/yNs2a0jRkYxy6191B2",
    id: "yNs2a0jRkYxy6191B2",
    images: { fixed_width: {} },
    import_datetime: "2018-03-28 18:38:53",
    is_sticker: 0,
    rating: "g",
    slug: "latenightseth-seth-meyers-lnsm-yNs2a0jRkYxy6191B2",
    source: "https://www.youtube.com/user/LateNightSeth/",
    source_post_url: "https://www.youtube.com/user/LateNightSeth/",
    source_tld: "www.youtube.com",
    title: "Fail Seth Meyers GIF by Late Night with Seth Meyers",
    trending_datetime: "0000-00-00 00:00:00",
    type: "gif",
    url:
      "https://giphy.com/gifs/latenightseth-seth-meyers-lnsm-yNs2a0jRkYxy6191B2",
    user: {
      avatar_url:
        "https://media1.giphy.com/avatars/latenightwithsethmeyers/wDIHpL5y74cm.jpg",
      banner_image: "",
      banner_url: "",
      profile_url: "https://giphy.com/latenightseth/",
      username: "latenightseth",
    },
    username: "latenightseth",
  },
  {
    analytics: {},
    analytics_response_payload: "",
    bitly_gif_url: "https://gph.is/2pLKLNW",
    bitly_url: "https://gph.is/2pLKLNW",
    content_url: "",
    embed_url: "https://giphy.com/embed/yNs2a0jRkYxy6191B2",
    id: "test",
    images: { fixed_width: {} },
    import_datetime: "2018-03-28 18:38:53",
    is_sticker: 0,
    rating: "g",
    slug: "latenightseth-seth-testing-lnsm-yNs2a0jRkYxy6191B2",
    source: "https://www.youtube.com/user/test/",
    source_post_url: "https://www.youtube.com/user/test/",
    source_tld: "www.youtube.com",
    title: "",
    trending_datetime: "0000-00-00 00:00:00",
    type: "gif",
    url:
      "https://giphy.com/gifs/latenightseth-seth-meyers-lnsm-yNs2a0jRkYxy6191B2",
    user: {
      avatar_url:
        "https://media1.giphy.com/avatars/latenightwithsethmeyers/wDIHpL5y74cm.jpg",
      banner_image: "",
      banner_url: "",
      profile_url: "https://giphy.com/latenightseth/",
      username: "latenightseth",
    },
    username: "latenightseth",
  },
];

describe("GiphyGrid", () => {
  it("renders GiphyGrid correctly with gifs prop", () => {
    const tree = renderer.create(<GiphyGrid gifs={mockGifs || []} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
