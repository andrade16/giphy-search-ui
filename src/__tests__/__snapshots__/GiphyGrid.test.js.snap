// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiphyGrid renders GiphyGrid correctly with gifs prop 1`] = `
<div
  className="giphy-grid-container"
>
  <ul
    className="MuiGridList-root grid-list"
    style={
      Object {
        "margin": -2,
      }
    }
  >
    <li
      className="MuiGridListTile-root"
      style={
        Object {
          "height": 164,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }
    >
      <div
        className="MuiGridListTile-tile"
      >
        <img
          alt="Fail Seth Meyers GIF by Late Night with Seth Meyers"
          onLoad={[Function]}
        />
      </div>
    </li>
    <li
      className="MuiGridListTile-root"
      style={
        Object {
          "height": 164,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }
    >
      <div
        className="MuiGridListTile-tile"
      >
        <img
          alt=""
          onLoad={[Function]}
        />
      </div>
    </li>
  </ul>
</div>
`;
